### PR TITLE
fix: onBecome(Un)Observed didn't trigger when using number as key of …

### PR DESCRIPTION
**Pull request by gespiton**
* [x] Added unit tests
* [x] Updated changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+
+# 5.13.0 / 4.13.0
```
Comment (FredyC) [inline CHANGELOG.md]: Move this under `5.13.1 / 4.13.1` as that version is not published yet.

Comment (gespiton) [reply CHANGELOG.md]: done.

Comment (FredyC) [reply CHANGELOG.md]: Um, I meant to merge it with the previous section, you have two with same versions there now :)

Comment (gespiton) [reply CHANGELOG.md]: thanks for your help, does this work now? @FredyC

Comment (FredyC) [reply CHANGELOG.md]: Ughhh ... I said to move this under 5.13.1 / 4.13.1, not to make another version :)

Comment (gespiton) [reply CHANGELOG.md]: I think I got it this time :P @FredyC

**Comment by FredyC** [comment]
It's merged in master manually as I had to tweak that changelog slightly, so I am closing it here, but it's there :)

**Comment by FredyC** [comment]
@gespiton Can you also prepare PR for MobX (into mobx4-master branch)? Since it's already in changelog like it's fixed there... :)

**Comment by gespiton** [comment]
> @gespiton Can you also prepare PR for MobX 4 (into `mobx4-master` branch)? Since it's already in the changelog, it should be actually fixed :)

ok

**Pull request closed without merging**